### PR TITLE
Add observedGeneration field to CR status

### PR DESCRIPTION
**Pull request by kabicin**

### CODE REVIEW

File: api/v1/runtimecomponent_types.go [hunk 1]
```diff
@@ -405,6 +405,9 @@ type RuntimeComponentStatus struct {
 	Binding *corev1.LocalObjectReference `json:"binding,omitempty"`
 
 	References common.StatusReferences `json:"references,omitempty"`
+
+	// The last generation of this RuntimeComponent instance observed by the operator.
```
Comment (leochr) [inline api/v1/runtimecomponent_types.go:409]: Suggestion: The generation identifier of this RuntimeComponent instance completely reconciled by the Operator.

Similar comment for the other instances

**Review by leochr — COMMENTED** [review]
@kabicin Thank you. Looks good. Just one minor comment for the definition

File: api/v1/runtimecomponent_types.go (hunk 1 above)
Comment (kabicin) [reply api/v1/runtimecomponent_types.go]: Updated the descriptions

File: api/v1/runtimeoperation_types.go [hunk 2]
```diff
@@ -42,6 +42,8 @@ type RuntimeOperationStatus struct {
 	// +listType=atomic
 	Conditions []OperationStatusCondition `json:"conditions,omitempty"`
 	Versions   StatusVersions             `json:"versions,omitempty"`
+	// The generation identifier of this RuntimeComponent instance completely reconciled by the Operator.
```
Comment (leochr) [inline api/v1/runtimeoperation_types.go:45]: RuntimeComponent --> RuntimeOperation

**Review by leochr — COMMENTED** [review]
@kabicin Thanks for the update. Just one comment about the name of the instance

**Review by leochr — APPROVED** [review]

**Pull request merged**